en/docs/develop-components/work-with-the-micro-integrator-runtime-in-choreo.md

```diff
@@ -61,10 +61,83 @@ Here are some of the benefits of adding third-party JARs to the Micro Integrator
+integration flow are available at runtime.
+
+In addition to `jar` libraries, you may also need to incorporate OSGi bundle JARs into your project. To do this,
+create another directory called `dropins` at the root of your project. Similarly, if the project root and GitHub
+repository root differ, place the `dropins` directory in the project root.
+
+Within the `dropins` directory, add the OSGi bundle JAR files. These bundles will be picked up by the OSGi runtime during
+deployment, enabling you to utilize their functionality within your Micro Integrator project.
 
 ![Libs directory](../assets/img/develop-components/micro-integrator/libs_dir.png)
 
+## Importing custom certificates to Micro Integrator
+
+The feature allows Choreo MI users to import certificates to the MI `client-truststore.jks` file. This is useful when
+users need to connect to servers or services that use SSL/TLS encryption and have self-signed or certificates issued by private CAs
+that are not trusted by default. By importing the certificate to the MI client-truststore.jks file, MI clients can
+establish a secure connection with the server or service without encountering SSL/TLS errors or warnings.
+
+Before importing the certificate, you need to obtain the certificate file. This file can be in
+the PEM or DER format. You can obtain the certificate file from the server or service provider or by exporting it from a
+web browser. Certificates should be added to the `/wso2mi/certs/` folder. Therefore, users must ensure that the mount path
+for the certificate file follows this path
+
+1. Go to the Component page and click on the **Deploy** page for the component you want to add the certificate to.
+2. Click on the **Configs and Secrets** tab.
+3. Click **Create**.
+4. Select **ConfigMap** as the **Config Type** and **File Mount** as the **Mount Type**.
+5. Click **Next**.
+6. Specify the following values as mount configurations:
+
+    | **Field**       | **Value**                                              |                                    
+    |-----------------|--------------------------------------------------------|
+    | **Config Name** | An appropriate name for the certificate.             |
+    | **Mount path**  | `/wso2mi/certs/<filename>`. For example, `/wso2mi/certs/certificate.crt` |
+
+7. Click **Upload File** and attach the certificate.
+8. Click **Create**.
+
+## Working with sensitive data using MI Secure Vault
+
+MI Secure Vault is a feature that allows users to securely store sensitive data, such as passwords and tokens, and
+provide access to that data to MI Integrations. This feature provides an added layer of security to your integrations by
+reducing the risk of accidental exposure of sensitive data.
+
+To use MI Secure Vault to work with sensitive data, follow these steps:
+
+1. Go to the Component page and click on the **Deploy** page.
```

Comment:
1. Select the component from the Choreo Console. 
2. From the left navigation, click **Deploy**.